tests/main/nss-modules/task.yaml

```diff
@@ -0,0 +1,315 @@
+            systemctl restart nslcd nscd
+
+        elif [ "$MODULE" = "sss" ]; then
+            # setup some defaults
+            envsubst < kdc.conf > /etc/krb5kdc/kdc.conf
+            envsubst < krb5.conf > /etc/krb5.conf
+            touch /etc/krb5kdc/kadm5.acl
+
+            # create the realm
+            kdb5_util create -s -P secretpassword
+
+            # restart services
+            systemctl restart krb5-kdc.service krb5-admin-server.service
+
+            # create the kerberos prinicpal
+            kadmin.local -q "addprinc -pw ${kerberos_principal_pw} ${nss_user}"
+            
+            # configure sssd to use ldap/kerberos
+            envsubst < sssd.conf > /etc/sssd/sssd.conf
+            chmod 0600 /etc/sssd/sssd.conf
+            chown root:root /etc/sssd/sssd.conf
+            systemctl restart sssd || {
+                systemctl status --lines 100 sssd
+                false
+            }
+        fi
+
+    elif [ "$MODULE" = "nis" ]; then
+        "$TESTSTOOLS"/lxd-state prepare-snap
+        "$TESTSTOOLS"/lxd-state launch --name server --remote ubuntu --image 24.04 --params '--vm'
+        lxc exec server -- apt update
+        lxc exec server -- useradd --uid 12346 --create-home ${nss_user}
+        lxc exec server -- apt install -y rpcbind nis
+        lxc exec server -- sh -c "echo \"$domain_name\" > /etc/defaultdomain"
+        lxc exec server -- domainname "$domain_name"
+        lxc exec server -- /etc/init.d/rpcbind restart
+        lxc exec server -- /etc/init.d/ypserv start
+
+        lxc file push ypinit.sh server/root/ypinit.sh
+        lxc exec server -- /root/ypinit.sh
+        lxc exec server -- systemctl restart ypserv
+        lxc exec server -- make -C /var/yp
+        lxc exec server -- sh -c "echo \"domain ${domain_name} server localhost\" >> /etc/yp.conf"
+        lxc exec server -- systemctl enable ypbind
+        lxc exec server -- systemctl start ypbind
+        
+        if ! [ -f /etc/defaultdomain ]; then
+            tests.cleanup defer rm /etc/defaultdomain
+        fi
+        domainname > /tmp/domainname.orig
+        tests.cleanup defer domainname "\"$(cat /tmp/domainname.orig)\""
+
+        echo "${domain_name}" > /etc/defaultdomain
+        domainname "${domain_name}"
+        echo "domain $domain_name" >> /etc/yp.conf
+        echo "ypserver $(lxc ls server -c 4 -f csv | awk '{print $1}')" >> /etc/yp.conf
+        if os.query is_ubuntu_ge 25.10; then
+            # the apparmor profile for hostname (which covers also execution of domainname)
+            # doesn't grant read permissions to /etc/defaultdomain, which is used in the
+            # ExecStartPre section of the ypbind service. Since the domain name is already set,
+            # comment out that line. Once https://gitlab.com/apparmor/apparmor/-/merge_requests/1814
+            # is deployed, this can be removed.
+            sed -i 's/^ExecStartPre=.*/# &/' /usr/lib/systemd/system/ypbind.service
```

Comment:
I think `ExecStartPre` can be removed by adding an empty one is a drop-in unit. Maybe easier than using sed.